--- awstats.orig/wwwroot/cgi-bin/awstats.pl	2013-09-25 14:50:40.069079621 +0400
+++ awstats/wwwroot/cgi-bin/awstats.pl	2013-09-25 14:51:31.001290632 +0400
@@ -18691,27 +18691,27 @@
 							}
 						}
 					}
-					else {
-						my $uarobot = 'no_user_agent';
+					#else {
+					#	my $uarobot = 'no_user_agent';
 
 						# It's a robot or at least a bad browser, we stop here
-						if ($Debug) {
-							debug(
-"  UserAgent not defined so it should be a robot, saved as robot 'no_user_agent'",
-								2
-							);
-						}
-						$_robot_h{$uarobot}++;
-						if ($pos_size>0){$_robot_k{$uarobot} += int( $field[$pos_size] );}
-						$_robot_l{$uarobot} = $timerecord;
-						if ( $urlwithnoquery =~ /$regrobot/o ) {
-							$_robot_r{$uarobot}++;
-						}
-						$countedtraffic = 4;
-						if ($PageBool) { $_time_nv_p[$hourrecord]++; }
-						$_time_nv_h[$hourrecord]++;
-						if ($pos_size>0){$_time_nv_k[$hourrecord] += int( $field[$pos_size] );}
-					}
+					#	if ($Debug) {
+					#		debug(
+#"  UserAgent not defined so it should be a robot, saved as robot 'no_user_agent'",
+					#			2
+					#		);
+					#	}
+					#	$_robot_h{$uarobot}++;
+					#	if ($pos_size>0){$_robot_k{$uarobot} += int( $field[$pos_size] );}
+					#	$_robot_l{$uarobot} = $timerecord;
+					#	if ( $urlwithnoquery =~ /$regrobot/o ) {
+					#		$_robot_r{$uarobot}++;
+					#	}
+					#	$countedtraffic = 4;
+					#	if ($PageBool) { $_time_nv_p[$hourrecord]++; }
+					#	$_time_nv_h[$hourrecord]++;
+					#	if ($pos_size>0){$_time_nv_k[$hourrecord] += int( $field[$pos_size] );}
+					#}
 				}
 			}
 		}
--- awstats.orig/wwwroot/cgi-bin/lib/browsers.pm	2013-09-25 17:05:06.044966289 +0400
+++ awstats/wwwroot/cgi-bin/lib/browsers.pm	2013-09-25 17:09:06.315439206 +0400
@@ -235,6 +235,7 @@
 'fdm',
 'javaws',
 'wget',
+'tpparser',
 'fget',
 'chilkat',
 'webdownloader\sfor\sx',
@@ -462,6 +463,7 @@
 'fdm','<a href="http://www.freedownloadmanager.org/" title="Browser home page [new window]" target="_blank">FDM Free Download Manager</a>',
 'javaws','Java Web Start',
 'wget','Wget',
+'tpparser', 'Travelsystem Php Parser',
 'fget','FGet',
 'chilkat', 'Chilkat',
 'webdownloader\sfor\sx','Downloader for X',
@@ -499,6 +501,7 @@
 'httrack','1',
 'webreaper','1',
 'wget','1',
+'tpparser', '1',
 'fget','1',
 'download\smaster','1',
 'webdownloader\sfor\sx','1',
