--- a/lib/redmine.rb	2020-11-21 22:22:26.751115318 +0300
+++ b/lib/redmine.rb	2020-11-21 22:39:03.792476944 +0300
@@ -265,8 +265,6 @@
             :html => {:class => 'icon icon-list'}
   menu.push :settings, {:controller => 'settings'},
             :html => {:class => 'icon icon-settings'}
-  menu.push :ldap_authentication, {:controller => 'auth_sources', :action => 'index'},
-            :html => {:class => 'icon icon-server-authentication'}
   menu.push :plugins, {:controller => 'admin', :action => 'plugins'}, :last => true,
             :html => {:class => 'icon icon-plugins'}
   menu.push :info, {:controller => 'admin', :action => 'info'}, :caption => :label_information_plural, :last => true,
